backend/api/Services/RobotService.cs

```diff
@@ -98,29 +99,142 @@ public async Task<Robot> CreateFromQuery(CreateRobotQuery robotQuery)
+                .SetProperty(p => p.Orientation.X, pose.Orientation.X)
+                .SetProperty(p => p.Orientation.Y, pose.Orientation.Y)
+                .SetProperty(p => p.Orientation.Z, pose.Orientation.Z)
+                .SetProperty(p => p.Orientation.W, pose.Orientation.W)
+                .SetProperty(p => p.Position.X, pose.Position.X)
+                .SetProperty(p => p.Position.Y, pose.Position.Y)
+                .SetProperty(p => p.Position.Z, pose.Position.Z)
+            );
+
+            robot = await robotQuery.FirstOrDefaultAsync();
+            ThrowIfRobotIsNull(robot, robotId);
+            NotifySignalROfUpdatedRobot(robot!, robot!.CurrentInstallation!);
 
-        public async Task<IEnumerable<Robot>> ReadAll()
+            return robot;
+        }
+        public async Task<Robot> UpdateRobotEnabled(string robotId, bool enabled)
         {
-            return await GetRobotsWithSubModels().ToListAsync();
+            var robotQuery = context.Robots.Where(robot => robot.Id == robotId).Include(robot => robot.CurrentInstallation);
+            var robot = await robotQuery.FirstOrDefaultAsync();
+            ThrowIfRobotIsNull(robot, robotId);
+
+            await VerifyThatUserIsAuthorizedToUpdateDataForInstallation(robot!.CurrentInstallation);
+
+            await robotQuery.ExecuteUpdateAsync(robots => robots.SetProperty(r => r.Enabled, enabled));
+
+            robot = await robotQuery.FirstOrDefaultAsync();
+            ThrowIfRobotIsNull(robot, robotId);
+            NotifySignalROfUpdatedRobot(robot!, robot!.CurrentInstallation!);
+
+            return robot;
         }
+        public async Task<Robot> UpdateCurrentMissionId(string robotId, string? currentMissionId)
+        {
+            var robotQuery = context.Robots.Where(robot => robot.Id == robotId).Include(robot => robot.CurrentInstallation);
+            var robot = await robotQuery.FirstOrDefaultAsync();
+            ThrowIfRobotIsNull(robot, robotId);
 
-        public async Task<Robot?> ReadById(string id)
+            await VerifyThatUserIsAuthorizedToUpdateDataForInstallation(robot!.CurrentInstallation);
+
+            await robotQuery.ExecuteUpdateAsync(robots => robots.SetProperty(r => r.CurrentMissionId, currentMissionId));
+
+            robot = await robotQuery.FirstOrDefaultAsync();
+            ThrowIfRobotIsNull(robot, robotId);
+            NotifySignalROfUpdatedRobot(robot!, robot!.CurrentInstallation!);
+
+            return robot;
+        }
+        public async Task<Robot> UpdateCurrentArea(string robotId, Area? area) { return await UpdateRobotProperty(robotId, "CurrentArea", area); }
+        public async Task<Robot> UpdateMissionQueueFrozen(string robotId, bool missionQueueFrozen)
         {
-            return await GetRobotsWithSubModels().FirstOrDefaultAsync(robot => robot.Id.Equals(id));
+            var robotQuery = context.Robots.Where(robot => robot.Id == robotId).Include(robot => robot.CurrentInstallation);
```

Comment:
Now that the functions are not all one-liners, could we maybe add a line between them to space them out?